internal/handlers/auth/okta.go

```diff
@@ -0,0 +1,84 @@
+package auth
+
+import (
+        "context"
+        "encoding/json"
+        "net/url"
+        "strings"
+
+        "github.com/mxschmitt/golang-url-shortener/internal/util"
+        "github.com/sirupsen/logrus"
+
+        "github.com/pkg/errors"
+        "golang.org/x/oauth2"
+)
+
+type oktaAdapter struct {
+        config *oauth2.Config
+}
+
+// NewOktaAdapter creates an oAuth adapter out of the credentials and the baseURL
+func NewOktaAdapter(clientID, clientSecret, endpointURL string) Adapter {
+
+        if endpointURL == "" {
+                logrus.Error("Configure Okta Endpoint")
```

Comment:
I would throw an error or on an other place, that the user has to configure this Endpoint URL. So that's a hard fail and the okta integration cannot configured incorrectly.

Reply:
Would `errors.New()` satisfy that, and should that also exist on line 55 or do you think this is enough?

Reply:
Hm, or in general it's okay. Logging should be fine for that to notice to the user, that he has to configure the Endpoint URL. I will merge it. Thanks for your work! 👍